Exercise 2 Multiplexer/multiplexer-rt/src/com/tridiumuniversity/multiplexer/BMux.java

```diff
@@ -0,0 +1,239 @@
+
+////////////////////////////////////////////////////////////////
+// Property "in0"
+////////////////////////////////////////////////////////////////
+  
+  /**
+   * Slot for the {@code in0} property.
+   * @see #getIn0
+   * @see #setIn0
+   */
+  public static final Property in0 = newProperty(Flags.SUMMARY, ((BBoolean)(BBoolean.DEFAULT)).getBoolean(), null);
+  
+  /**
+   * Get the {@code in0} property.
+   * @see #in0
+   */
+  public boolean getIn0() { return getBoolean(in0); }
+  
+  /**
+   * Set the {@code in0} property.
+   * @see #in0
+   */
+  public void setIn0(boolean v) { setBoolean(in0, v, null); }
+
+////////////////////////////////////////////////////////////////
+// Property "in1"
+////////////////////////////////////////////////////////////////
+  
+  /**
+   * Slot for the {@code in1} property.
+   * @see #getIn1
+   * @see #setIn1
+   */
+  public static final Property in1 = newProperty(Flags.SUMMARY, ((BBoolean)(BBoolean.DEFAULT)).getBoolean(), null);
+  
+  /**
+   * Get the {@code in1} property.
+   * @see #in1
+   */
+  public boolean getIn1() { return getBoolean(in1); }
+  
+  /**
+   * Set the {@code in1} property.
+   * @see #in1
+   */
+  public void setIn1(boolean v) { setBoolean(in1, v, null); }
+
+////////////////////////////////////////////////////////////////
+// Property "in2"
+////////////////////////////////////////////////////////////////
+  
+  /**
+   * Slot for the {@code in2} property.
+   * @see #getIn2
+   * @see #setIn2
+   */
+  public static final Property in2 = newProperty(Flags.SUMMARY, ((BBoolean)(BBoolean.DEFAULT)).getBoolean(), null);
+  
+  /**
+   * Get the {@code in2} property.
+   * @see #in2
+   */
+  public boolean getIn2() { return getBoolean(in2); }
+  
+  /**
+   * Set the {@code in2} property.
+   * @see #in2
+   */
+  public void setIn2(boolean v) { setBoolean(in2, v, null); }
+
+////////////////////////////////////////////////////////////////
+// Property "in3"
+////////////////////////////////////////////////////////////////
+  
+  /**
+   * Slot for the {@code in3} property.
+   * @see #getIn3
+   * @see #setIn3
+   */
+  public static final Property in3 = newProperty(Flags.SUMMARY, ((BBoolean)(BBoolean.DEFAULT)).getBoolean(), null);
+  
+  /**
+   * Get the {@code in3} property.
+   * @see #in3
+   */
+  public boolean getIn3() { return getBoolean(in3); }
+  
+  /**
+   * Set the {@code in3} property.
+   * @see #in3
+   */
+  public void setIn3(boolean v) { setBoolean(in3, v, null); }
+
+////////////////////////////////////////////////////////////////
+// Type
+////////////////////////////////////////////////////////////////
+  
+  @Override
+  public Type getType() { return TYPE; }
+  public static final Type TYPE = Sys.loadType(BMux.class);
+
+/*+ ------------ END BAJA AUTO GENERATED CODE -------------- +*/
+
+
+  @Override
+  public void changed(Property property, Context context) {
+    super.changed(property, context);
+
+    if(!out.equals(property)){

```

Comment:
It would be slightly better to check if `property` is equal to any of `s0`, `s1`, `in0`, `in1`, `in2`, or `in3` instead of checking if `property` is not `out`. This is more verbose, but it would make this component more future-proof.

For instance, suppose another developer wanted to add a new property, `notOut`, that is always the opposite of `out`. The code in this `if` block would be run whenever `notOut` changes unless that developer updated the `if` condition.